lib/ansible/modules/cloud/amazon/ec2_group.py

```diff
@@ -284,6 +295,27 @@ def get_target_from_rule(module, ec2, rule, name, group, groups, vpc_id):
     return group_id, ip, target_group_created
 
 
+def ensure_tags(ec2, group, tags, changed, module):
+    try:
+        resource_id = group.id
+        cur_tags = group.tags
+        if cur_tags == tags:
```

Comment:
I don't think there's a feature that will do this so yeah, I think we should probably iterate through the dict and use to_text to ensure all the values are strings. Instead of modifying with each value with 'str()' use 'to_text()' (you'll also need to add `from ansible.module_utils._text import to_text` to the imports). Besides that, this looks great. Thanks for the time you've spent on it :)